### PR TITLE
adding third derivative of massive nu pressure

### PR DESCRIPTION
**Pull request by simonepeirone**

### CODE REVIEW

File: modules.f90 [hunk 1]
```diff
@@ -1879,6 +1882,35 @@ function Nu_pidotdot(am,adotoa,Hdot,presnu,presnudot) result (presnudotdot)
...
+        else
+
+            d=log(am/am_min)/dlnam+1._dl
+            i=int(d)
+            d=d-i
+
+            presnudotdotdot = dddp1(i)+d*(ddddp1(i)+d*(3._dl*(dddp1(i+1)-dddp1(i))-2._dl*ddddp1(i) &
+                -ddddp1(i+1)+d*(ddddp1(i)+ddddp1(i+1)+2._dl*(dddp1(i)-dddp1(i+1)))))
+
+            presnudotdotdot = 0._dl !IW
```
Comment (simonepeirone) [inline modules.f90]: not sure what to add here. Where should I look at to implement this cubic spline interpolation? @EFTCAMB

**Comment by simonepeirone** [comment]
it is not clear to me what to add at line 1910 of modules.f90. where is the definition of the cubic spline we are using?

File: eftcamb/03_abstract_EFTCAMB_cache.f90 [hunk 2]
```diff
@@ -145,6 +146,21 @@ function Nu_pidotdot_Wrapper( am, adotoa, Hdot, presnu, presnudot )
             real(dl)              :: Nu_pidotdot_Wrapper !< output value of the second time derivative of neutrino pressure
         end function Nu_pidotdot_Wrapper
         !----------------------------------------------------------------------------------------
+        !> Wwrapper to the function that computes the background massive neutrinos
```
Comment (mraveri) [inline eftcamb/03_abstract_EFTCAMB_cache.f90]: Typo Wwrapper -> Wrapper

File: modules.f90 (hunk 1 above)
Comment (mraveri) [reply modules.f90]: This should be correct. Did you check numerically that the result was expected?

**Review by mraveri — COMMENTED** [review]
A couple of things to fix. After that PR can be merged.

**Comment by simonepeirone** [comment]
now it the third derivative looks correct
[1_pressure_ddd.pdf](https://github.com/EFTCAMB/EFTCAMB/files/947432/1_pressure_ddd.pdf)
[2_pressure_deviation.pdf](https://github.com/EFTCAMB/EFTCAMB/files/947433/2_pressure_deviation.pdf)

the error is always <1%, like for the second derivative

**Comment by mraveri** [comment]
Is the agreement 2%?

**Comment by simonepeirone** [comment]
Indeed

**Comment by mraveri** [comment]
This sounds a bit too much to be just interpolation noise.

Could it be due to some missing terms?

**Comment by simonepeirone** [comment]
yes, it could definitely be some missing terms.

Anyway this is the error we have on the second derivative, which is the same order of magnitude
[2_second_derivative.pdf](https://github.com/EFTCAMB/EFTCAMB/files/950791/2_second_derivative.pdf)

**Pull request merged**